frontend/src/components/core/Sidebar/SidebarNav.tsx

```diff
@@ -57,7 +57,18 @@ const SidebarNav = ({
     return null
   }
 
-  const [expanded, setExpanded] = useState(false)
+  // Check localStorage to see if the user has a preference set
+  const isLocalStorageSet = localStorage.getItem("navExpanded") !== null
+  const shouldNavExpand =
+    isLocalStorageSet && localStorage.getItem("navExpanded") === "true"
+  const pageLimit = shouldNavExpand === true ? -1 : 6
```

Comment:
It seems weird to me to have a pageLimit be `-1`... could you explain why this is the case versus using 0 for example?

Reply:
> could you explain why this is the case versus using 0 for example?

Yeah! `0` wouldn't work because we're using this variable to slice the array depending on the conditions above, and if we set it to zero, then you'll end up with something like: `appPages.slice(0, 0)`, which might not work as expected.

Your comment though, sparked a better solution which is using `appPages.length` as the upper bound in those cases: I've fixed it on [eb2737f](https://github.com/streamlit/streamlit/pull/6050/commits/eb2737fe42c76d673771f7239ed34e248614c74e), let me know if you think that's more elegant!